packages/services/usage-ingestor/src/processor.ts

```diff
@@ -136,10 +156,29 @@ export function createProcessor(config: { logger: ServiceLogger }) {
             }),
           );
         }
+
+        for (const [hash, processedOperations] of processedOperationWaitingForInsert.entries()) {
+          const operationType = hashToOperationTypeMapping.get(hash);
+          if (operationType === undefined) {
+            logger.warn(`Operation type not found for hash. Skipping insert. hash(=%s)`, hash);
+            continue;
+          }
+
+          if (operationType === 'subscription') {
+            serializedSubscriptionOperations.push(
+              ...processedOperations.map(record => stringifySubscriptionOperation(record)),
+            );
+          } else {
+            serializedOperations.push(
+              ...processedOperations.map(record => stringifyQueryOrMutationOperation(record)),
+            );
+          }
+        }
```

Comment:
I delayed the stringification until we determined the operation type. We could also from the client send the operation type.

E.g. if we want also to support live queries later on/immediately, these would also be stored within `subscription_operations`.

So, I am tempted to instead send a type with every operation:
- `subscription`: graphql live query, graphql subscription
- `request`: graphql mutation, graphql query

As a legacy fallback, we can always use `request`, as today we don't send any live query/graphql subscription stuff to hive.

Reply:
done via [`2d8fdd0` (#4228)](https://github.com/kamilkisiela/graphql-hive/pull/4228/commits/2d8fdd03a35c01119d27997c3e364cc214e4089d)